### PR TITLE
Bug 1968019: Bump drain timeout to 1h

### PR DESCRIPTION
**Pull request by kikisdeliveryservice**
As per retro and subsequent follow-up conversation, make a short term bump to the pool drain degradation to 1hour since the existing times are too short for many users and cause premature erroring/degrading.

In addition:
Refactor drain() since exponential backoff is not suitable for these longer durations. 
Refactor the metric and reduce cardinality, and remove 15min delay on alert.

### CODE REVIEW

File: pkg/daemon/drain.go
Comment (dhellmann) [inline pkg/daemon/drain.go:64]: I'm still relatively new to go, but it looks like if this error case is triggered nothing is going to stop `drainer()`. Is that right, or is there something subtle that I'm not seeing?

Comment (kikisdeliveryservice) [reply pkg/daemon/drain.go:72]: that was uhh.. not right :)

Comment (dhellmann) [reply pkg/daemon/drain.go:72]: This looks better. :-) I would probably move line 76 up to happen first in this block, but the order may not actually matter.

Comment (kikisdeliveryservice) [reply pkg/daemon/drain.go:73]: sure, gave it a whirl :smile:

File: pkg/daemon/drain.go
Comment (dhellmann) [inline pkg/daemon/drain.go:51]: Does something need to close this channel if there's a timeout? Or will the garbage collector do that?

File: pkg/daemon/drain.go
Comment (dhellmann) [inline pkg/daemon/drain.go:59]: Should this log message include the delay? As written, it implies that the retry might start immediately.

**Review by dhellmann — COMMENTED** [review]
The logic looks good. Another go noob question about the gc and one nit about a log message.

File: pkg/daemon/drain.go
Comment (kikisdeliveryservice) [reply pkg/daemon/drain.go:59]: No it's fine as is and consistent with how it was done before.

File: pkg/daemon/drain.go
Comment (darkmuggle) [reply pkg/daemon/drain.go:51]: The channel will be GC'd when `drain()` exists since the scope of `done` is contained entirely in `drain()`

**Comment by kikisdeliveryservice** [comment]
I think I might attach this to a different bug, removing the WIP but adding hold to sort that out. Otherwise, this is tested and ready to go.

/hold

File: pkg/daemon/drain.go
Comment (darkmuggle) [inline pkg/daemon/drain.go:72]: This is a cleaner way, IMO, than an exponential back-off due to length of time involved. The usefulness of an exponential back-off diminishes when the time scales move much beyond 5min. Further, you'll have the ability to tweak the timeouts easily.

**Review by darkmuggle — APPROVED** [review]
IMO, this is easier to ready than an exponential back-off. 

FWIW, /lgtm

File: pkg/daemon/drain.go
Comment (dhellmann) [reply pkg/daemon/drain.go:51]: Thanks for explaining!

Comment (kikisdeliveryservice) [reply pkg/daemon/drain.go:51]: thanks for your review @dhellmann !!

**Comment by kikisdeliveryservice** [comment]
Since this wouldn't close Bugzilla 1952694 fully created a new bug so we can backport this as an intermediate fix and to get signal.

cc: @sdodson

File: pkg/daemon/drain.go
Comment (kikisdeliveryservice) [reply pkg/daemon/drain.go:72]: agree was much easier when testing to tweak and much clearer/more predictable in logs etc..

thanks for your help!!

**Comment by kikisdeliveryservice** [comment]
seem to be hitting https://bugzilla.redhat.com/show_bug.cgi?id=1968079 in gcp-op

**Comment by kikisdeliveryservice** [comment]
This has been ready since last week & has been reviewed - can someone please LGTM

**Pull request merged**
